mantidproject/mantid
# Indirect - Diffraction Diffspec - Prevent check for 'runs' equality unless SumFiles is false

In the ISISIndirectDiffractionReduction algorithm, it is usually the case that the number of runs in the samples file and the vanadium runs file need to be equal. This is not true when SumFiles is set to true/selected.

Added a falsity check for the SumFiles property before checking whether the number of runs in the samples file is equal to the number of runs in the vanadium file.

**To test:**
1. Access the Diffraction interface by navigating to Interfaces > Indirect > Diffraction.
2. Select OSIRIS as the Instrument.
3. Ensure 'diffspec' is selected in the Reflection drop-down menu.
4. Select an input *_red.nxs (sample) file.
5. Ensure that the 'Sum Files' checkbox is checked.
6. Ensure that the checkbox 'Use Vanadium File' is checked.
7. Select a vanadium range with a different number of runs than the input sample range.
8. Run the algorithm and ensure it runs successfully without throwing an error/exception.

Sample Run Numbers: 10203-10204
Vanadium Run Numbers: 10156

Fixes #20031. 

**Release Notes** 

Added an explanation of the bug fix to the release notes.

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
